mlflow/models/utils.py

```diff
@@ -470,24 +485,61 @@ def _is_sparse_matrix(x):
         elif isinstance(pfInput, pd.DataFrame):
             new_pfInput = {}
             for col_name, tensor_spec in zip(input_schema.input_names(), input_schema.inputs):
-                new_pfInput[col_name] = _enforce_tensor_spec(
-                    np.array(pfInput[col_name], dtype=tensor_spec.type), tensor_spec
-                )
+                pd_series = pfInput[col_name]
+                if np.isscalar(pd_series[0]):
+                    if tensor_spec.shape != (-1,):
+                        raise MlflowException(
+                            f"The input pandas dataframe column '{col_name}' contains scalar "
+                            "values, which requires the shape to be (-1,), but got tensor spec "
+                            f"shape of {tensor_spec.shape}.",
+                            error_code=INVALID_PARAMETER_VALUE,
+                        )
+                    new_pfInput[col_name] = _enforce_tensor_spec(
+                        np.array(pd_series, dtype=tensor_spec.type), tensor_spec
+                    )
+                else:
+                    # If the pandas column contains list type values,
```

Comment:
Is this really true? Is `not np.isscalar(pd_series[0])` equivalent to `pd_series[0] is a list`?